### PR TITLE
Spredfast 2.0 updates

### PR DESCRIPTION
**Pull request by kelsy**
This adds post_message and update_message which are tweaked versions of the current API endpoints. Now we can set client.body using the serializer and then client.post_message to post that message body to the publishing endpoint. This currently doesn't work for asset posts yet. :(

### CODE REVIEW

**Comment by phantummm** [comment]
Looks good :+1:

File: lib/spredfast/post_image.rb [hunk 1]
```diff
@@ -95,19 +101,41 @@ def put
 					response["body"] = JSON.parse(log.body) if log.body
 				end
 				response
-			rescue =>e
-				puts "EXCEPTION: #{e.response}"
-				puts "ERROR message: #{e.message}"
+			rescue => e
+				JSON.parse(e.response)
```
Comment (thoughtpunch) [inline lib/spredfast/post_image.rb]: @kelsy / @phantummm this is throwing an error after multiple image post updates. Looks like we need to handle the case where we get non-standard/bad/nil responses

```
SPREDFAST API UPDATE IMAGE: https://api.spredfast.com/VPC1/v1/company/221/initiative/1513/message/18172495
Completed 500 Internal Server Error in 6479ms

NoMethodError - undefined method `response' for #<JSON::ParserError:0x007f8ce12a2bd0>:
  spredfast (0.3.0) lib/spredfast/post_image.rb:105:in `rescue in put'
  spredfast (0.3.0) lib/spredfast/post_image.rb:92:in `put'
  spredfast (0.3.0) lib/spredfast/post_image.rb:38:in `update_image'
  lib/publisher/spredfast/opal_client.rb:78:in `update_asset_post'
  lib/publisher/spredfast/opal_client.rb:41:in `deliver_update'
  lib/publisher/spredfast/opal_client.rb:26:in `deliver'
  opal_apps/app_moments/app/jobs/content_delivery_worker.rb:23:in `perform'
```

**Comment by phantummm** [comment]
@kelsy / @thoughtpunch Alright, so I did a bunch of cleanup on the `PostImage` client, and everything from `Spredfast::OpalClient` to `DeliveryRecord` seems to be working a-ok.

**Comment by phantummm** [comment]
Merging PR, which reflects the current version (`0.3.2`) pushed to `gems.workwithopal.com` and in use on prod.

**Comment by andrewvy** [comment]
:100:

**Pull request merged**